onadata/libs/serializers/data_serializer.py

```diff
@@ -303,3 +303,48 @@ def create(self, validated_data):
                                      validated_data['id_string'])
 
         return instance
+
+
+class FLOIPSubmissionSerializer(SubmissionSuccessMixin,
+                                serializers.Serializer):
+    """
+    FLOIP SubmmissionSerializer - Handles a row of FLOIP specification format.
+    """
+    def validate(self, attrs):
+        data = self.context['request'].data
+        error_msg = None
+
+        if not isinstance(data, list):
+            error_msg = u'Invalid format. Expecting a list.'
+        else:
+            for row_i in xrange(len(data)):
+                if len(data[row_i]) != 6:
+                    error_msg = u"All rows must have 6 values, "\
+                                u"row %s does not." % row_i
+                    break
```

Comment:
Could we state what the 6 values should be?

Also, you should probably apply `ugettext()` here for this to be available in translation files.
```python
_("All rows must have 6 values, row %(row)s does not." % {'row': row_i})
```
Applying on the variable should work when it is being translated but the string will not be included in the translation files.